packages/js/product-editor/src/components/manage-download-limits-modal/manage-download-limits-modal.tsx

```diff
@@ -0,0 +1,216 @@
+/**
+ * External dependencies
+ */
+import { FormEvent } from 'react';
+import classNames from 'classnames';
+import { useInstanceId } from '@wordpress/compose';
+import { createElement, useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	BaseControl,
+	Button,
+	Modal,
+	// @ts-expect-error `__experimentalInputControl` does exist.
+	__experimentalInputControl as InputControl,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { ManageDownloadLimitsModalProps } from './types';
+import { useNumberInputProps } from '../../hooks/use-number-input-props';
+
+const DOWNLOAD_LIMIT_MIN = 1;
+const DOWNLOAD_EXPIRY_MIN = 1;
+
+export function ManageDownloadLimitsModal( {
+	initialValue,
+	onSubmit,
+	onClose,
+}: ManageDownloadLimitsModalProps ) {
+	const [ downloadLimit, setDownloadLimit ] = useState< string >(
+		initialValue.downloadLimit ? String( initialValue.downloadLimit ) : ''
```

Comment:
When the value is set to `unlimited` it is set to `-1`, which shows up in the form (see pic below).
We should set the value here to empty string when the `downloadLimit` is `-1`, so the placeholder shows correctly. Same goes for `downloadExpiry`.
<img width="708" alt="Screenshot 2023-10-18 at 11 30 51 AM" src="https://github.com/woocommerce/woocommerce/assets/2240960/0b19fb0e-a517-4db6-932a-7f2728ccb3f4">

Reply:
done -> https://github.com/woocommerce/woocommerce/pull/40797/commits/e5b244c93ba4a58516582a2806917b8349f34914